flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/sink/RowDataTaskWriterFactory.java

```diff
@@ -59,6 +60,16 @@ public RowDataTaskWriterFactory(Table table,
                                   FileFormat format,
                                   List<Integer> equalityFieldIds,
                                   boolean upsert) {
+    this(table, table.properties(), flinkSchema, targetFileSizeBytes, format, equalityFieldIds, upsert);
```

Comment:
The writer should be passed the table properties, without merging in the options. I just looked at the file appender and that appears to be the correct behavior.

Reply:
This is mainly because previously we didn't limit the range of configurations that users could override.  Options may contain other options that are not listed in `FlinkWriteOptions`.  This should no longer be required at this time.